src/services/deep-research/DeepResearchService.ts

```diff
@@ -0,0 +1,270 @@
+import { createOpenAI } from "@ai-sdk/openai"
+import FirecrawlApp from "@mendable/firecrawl-js"
+import { generateObject, LanguageModel } from "ai"
+import { z } from "zod"
+import pLimit from "p-limit"
+
+import { ExtensionMessage } from "../../shared/ExtensionMessage"
+
+import { generateSerpQueries, processSerpResult } from "./utils/serp"
+import { ClineProvider } from "../../core/webview/ClineProvider"
+
+export type ResearchProgress = {
+	currentDepth: number
+	totalDepth: number
+	currentBreadth: number
+	totalBreadth: number
+	currentQuery?: string
+	totalQueries: number
+	completedQueries: number
+}
+
+export type ResearchResult = {
+	learnings: string[]
+	visitedUrls: string[]
+}
+
+const firecrawl = new FirecrawlApp({
+	apiKey: process.env.FIRECRAWL_KEY ?? "",
+	apiUrl: process.env.FIRECRAWL_BASE_URL,
+})
+
+export class DeepResearchService {
+	private providerRef: WeakRef<ClineProvider>
+	private model: LanguageModel
+	private initialQuery?: string
+	private combinedQuery?: string
+	private questions: string[] = []
+	private answers: string[] = []
+
+	constructor(
+		clineProvider: ClineProvider,
+		public readonly modelId = "o3-mini",
+		public readonly breadth = 4,
+		public readonly depth = 2,
+		public readonly concurrency = 2,
+	) {
+		this.providerRef = new WeakRef(clineProvider)
+
+		const modelProvider = createOpenAI({
+			apiKey: process.env.OPENAI_API_KEY,
+			baseURL: "https://api.openai.com/v1",
+		})
+
+		this.model = modelProvider(modelId, {
+			// reasoningEffort: "medium",
+			structuredOutputs: true,
+		})
```

Comment:
@cte  can we set api config profile for this ?, like enhance prompt ?

Reply:
Yep, for sure; haven't gotten to the settings piece yet.